hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/ComputeStageImplBase.java

```diff
@@ -354,22 +353,23 @@
     }
 
     @Nonnull
+    @SuppressWarnings("unchecked")
     <K, RET> RET attachPartitionedCustomTransform(
             @Nonnull String stageName,
             @Nonnull ProcessorMetaSupplier procSupplier,
             @Nonnull DistributedFunction<? super T, ? extends K> partitionKeyFn
-
     ) {
-        return attach(partitionedCustomProcessorTransform(stageName, transform, procSupplier, partitionKeyFn),
+        DistributedFunction adaptedKeyFn = fnAdapter.adaptKeyFn(partitionKeyFn);
```

Comment:
There are few more instances where the keyFn is not adapted, for example in `BatchStageWithKeyImpl.aggregate*`

Reply:
Batch stages don't need adaptation, it's only for timestamped items